sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala

```diff
@@ -126,6 +127,27 @@ private[sql] trait SQLTestUtils
   }
 
   /**
+   * Creates the requested number of temporary path (without creating the actual file/directory),
```

Comment:
I wanted to create a multi-path version of an existing withTempPath function. That function returns a *valid*, but non-existent path. The comments and path.delete() part of the implementation came from there.

/**
   * Generates a temporary path without creating the actual file/directory, then pass it to `f`. If
   * a file/directory is created there by `f`, it will be delete after `f` returns.
   *
   * @todo Probably this method should be moved to a more general place
   */
  protected def withTempPath(f: File => Unit): Unit = {
    val path = Utils.createTempDir()
    path.delete()
    try f(path) finally Utils.deleteRecursively(path)
  }